qml/CountryPage.qml

```diff
@@ -26,8 +28,77 @@ BasePage {
     forwardButtonSourceComponent: forwardButton
 
     property variant currentRegion: null
+    property string mccCountryCode: ""
     property int currentRegionIndex: -1
 
+    Item {
+        id: mapper
+
+        ListModel {
+            id: dataModel
+        }
+
+        function loadData() {
+            var xhr = new XMLHttpRequest
+            var jsonSource = ""
+            xhr.open("GET", "file:///etc/palm/mccInfo.json")
+            xhr.onreadystatechange = function () {
+                if (xhr.readyState === XMLHttpRequest.DONE)
+                    updateJSONModel(xhr.responseText)
+            }
+            xhr.send()
+        }
+
+        function updateJSONModel(source) {
+            dataModel.clear()
+
+            if (source === "") {
+                return
+            }
+
+            var objectArray = JSON.parse(source)
+            for (var key in objectArray) {
+                var jo = objectArray[key]
+                dataModel.append(jo)
+            }
+
+            handleUpdatedModel()
+        }
+
+        property variant __currentCall: null
+
+        function handleUpdatedModel() {
+            if (__currentCall) {
+                __currentCall.cancel()
+            } else {
+                __currentCall = connectNetwork.subscribe(JSON.stringify({
+                                                                            subscribe: true
+                                                                        }))
+            }
+        }
+    }
+
+    LunaService {
+        id: connectNetwork
+        name: "org.webosports.app.firstuse"
+        usePrivateBus: true
+        service: "luna://com.palm.telephony"
+        method: "networkIdQuery"
+
+        onResponse: function (message) {
+            var response = JSON.parse(message.payload)
+            var mcc = parseInt(response.extended.mccmnc.substring(0, 3))
+            for (var n = 0; n < dataModel.count; n++) {
+                var entry = dataModel.get(n)
+                if (mcc === entry.mcc) {
+                    mccCountryCode = entry.CountryCode.toLowerCase()
+                    console.log("We found countrycode: "+mccCountryCode+" for MCC: "+response.extended.mccmnc.substring(0, 3))
+                }
+            }
+            getRegions()
```

Comment:
When this a separate component just send a signal here "done" or assign a property which the outside can then listen for.